tests/perf/test_benchmark.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import cProfile
+
+import os
+from collections import namedtuple
+
+import pytest
+import re
+import torch
+
+import pyro
+import pyro.distributions as dist
+from pyro.distributions.testing import fakes
+from pyro.infer import SVI
+import pyro.optim as optim
+from pyro.infer.mcmc.hmc import HMC
+from pyro.infer.mcmc.mcmc import MCMC
+from pyro.infer.mcmc.nuts import NUTS
+
+
+Model = namedtuple('TestModel', ['model', 'model_args', 'model_id'])
+
+
+TEST_MODELS = []
+MODEL_IDS = []
+ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir))
+PROF_DIR = os.path.join(ROOT_DIR, ".benchmarks")
+if not os.path.exists(PROF_DIR):
+    os.makedirs(PROF_DIR)
+
+
+def register_model(**model_kwargs):
+    def register_fn(model):
+        model_id = model_kwargs.pop("id")
```

Comment:
I initially had an id generation function - see https://github.com/uber/pyro/pull/858/commits/077f009dc0d39442e7b9195196ba6dc9f508e01c. The problem with that are twofold:
 - It mucks up the report that is published in the end, because there is a certain way pytest generates names for parameterized tests, and the resulting names were too long. e.g. `test_benchmark[bernoulli_beta_hmc::kernel=NUTS_step_size=0_02_num_steps=10]`, and the resulting tabbed output would get completely mis-aligned. There are way to shorten the output, but not by much, and it is just so much easier for the test designer to create a short id for their test. :)
 - The other reason is to know exactly what ID a test has when you want to profile it. Since there are multiple parameterized tests per test, having an explicit name makes it easy for the user to just paste that when running the profiler via `make profile models='xyz'` (e.g. maybe the user just wants to run snakeviz on the HMC test).